smac/facade/smac_facade.py

```diff
@@ -207,44 +253,64 @@ def __init__(self,
 
         # initial EPM
         types, bounds = get_types(scenario.cs, scenario.feature_array)
+        model_def_kwargs = {
+            'types':types,
```

Comment:
I think we should add a check here that types and bounds and the feature_array and the seed are not overwritten from the outside.